CMakePresets.json

```diff
@@ -6,6 +6,10 @@
       "description": "Base Build Configuration",
       "binaryDir": "${sourceDir}/out/build/${presetName}",
       "installDir": "${sourceDir}/out/install/${presetName}",
+      "generator": "Ninja",
```

Comment:
This forces a ninja requirement on the build. Do we have to, in which case can that be added to the documentation, or can the generator be optional (and document how to specify it)?

```
CMake Error at CMakeLists.txt:3 (project):
  Running

   '/CommonExtensions/Microsoft/CMake/Ninja/ninja.exe' '--version'

  failed with:

   The system cannot find the file specified
 ```

Reply:
Made optional, but I have kept the search for `ninja.exe`.